### ISSUE TITLE
libgit2sharp-ssh :user cancelled certificate check exception

### ISSUE THREAD

**Issue by rotemwolfovich** [libgit2/libgit2sharp#1600]
You are opening a _bug report_ against the LibGit2Sharp project: we
use GitHub Issues for tracking bug reports and feature requests.  If
you have a question about an API or usage, please ask on StackOverflow:
http://stackoverflow.com/questions/tagged/libgit2sharp.

Otherwise, to report a bug, please fill out the reproduction steps
(below) and delete these introductory paragraphs.  Thanks!

### Reproduction steps
win7/win 10 using the libgit2sharp-ssh 1.0.2 
clone repository  using app written in visual studio  2015 update 3 
throw exception : "user cancelled certificate check" 
(when cloning using gitBash not with code works fine   )
### Expected behavior
clone via https under corporate proxy 
### Actual behavior
my app was using libgit2sharp version 0.22.0  and everything was working fine, also under corporate proxy . 
after updating the dll to libgit2sharp-ssh i get the exception :  "user cancelled certificate check" when using https/http (ssh option works ok)  
### Version of LibGit2Sharp (release number or SHA1)
ibgit2sharp-ssh 1.0.2
### Operating system(s) tested; .NET runtime tested
  C# .net 4.5 

is libgit2sharp-ssh not support proxy ? 
if not, how can i  implement connection to git with https/http and ssh with  corporate proxy .

**Comment by ethomson** [comment]
I'm not familiar with LibGit2Sharp-SSH.  Looks like this is something that somebody else has cooked up, and it's based on a very old LibGit2Sharp.  I would encourage you to reach out to them.

**Comment by rotemwolfovich** [comment]
there is some  version of libgit2sharp that support ssh ?  except LibGit2Sharp-SSH

**Comment by ethomson** [comment]
Not as a nuget package.  We do not provide builds of libssh2 - you would need to build it yourself.

"User cancelled certificate check" means that there's an untrusted certificate.  You should either use a certificate on your server that is trusted by your system, or provide a certificate hook check to accept it.  Technically speaking, you can also turn off all git's certificate checking, but I don't recommend it.

**Comment by rotemwolfovich** [comment]
ok , is it possible that if i am working under company proxy i will get this exception  ? 
because when i tried to clone in a different network ( home network for example ) everything was working fine .

**Comment by ethomson** [comment]
Yes, if the corporate proxy is presenting a self-signed certificate, or one that was not issued with a trusted cert chain.  This is common in corporate proxies.

**Comment by rotemwolfovich** [comment]
thank you for clearing this up . 
do you know if there is a plan to  add the ssh support ?  the latest status  i found was from 2016 (in this issue :  Introduce SSH functionality #1072) 
do you know on which version of libgit2sharp  the libgit2sharp-ssh is base on ?
